### ISSUE TITLE
Enforce pure YAML syntax

### ISSUE THREAD

**Issue by AMeng** [willthames/ansible-lint#164]
# Feature request

# Desired Behavior

Add a rule to enforce pure YAML. There are a number of benefits to this (linked below). The rule would disallow the use of `key=value` in a task. For example, instead of writing a `mount` directive like this:

```yaml
---
- name: Ensure the floppy disk is mounted.
  mount: "name=/mnt/floppy src=/dev/fd0 fstype=fuse opts=allow_other,user=nobody,nonempty dump=0 passno=0 state=mounted"
```

It should be written like this:

```yaml
---
- name: Ensure the floppy disk is mounted.
  mount:
    name: /mnt/floppy
    src: /dev/fd0
    fstype: fuse
    opts: "allow_other,user=nobody,nonempty"
    dump: 0
    passno: 0
    state: mounted
```

This could certainly be off by default, but in my opinion all Ansible roles should be written this way. Here are some useful links justifying this choice:

 * [Why you should use YAML as YAML with Ansible](http://blog.bandwidth.com/why-you-should-use-yaml-as-yaml-with-ansible/)
 * [OpenShift Ansible Best Practices](https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#Ansible-files-SHOULD-NOT-use-JSON-use-pure-YAML-instead)
 * [YAML Best Practices for Ansible Playbooks](http://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks)

**Comment by willthames** [comment]
I've written an ansible-review rule that does this - I'll see if I can make it an ansible-lint rule. 

I may well add it to an extras rule set rather than the defaults, not sure how popular new rules always are (I was happy to add a few new ones with v3.0)

**Comment by willthames** [comment]
Looking at how ansible-lint currently runs checks, this is much easier to do with ansible-review (I could add a fourth matcher that just runs against the content of a file, but this is already implemented in ansible-review)

https://github.com/willthames/ansible-review/blob/master/examples/standards.py#L21
https://github.com/willthames/ansible-review/blob/master/examples/standards.py#L233

ansible-review can also run whatever lint checks you require against certain classes of file.

**Comment by willthames** [comment]
ansible-lint 3.1.0 released with this change
